### PR TITLE
Maticjs v3 update

### PR DESCRIPTION
**Pull request by py-zoid**
Update matic js to v3

### CODE REVIEW

File: src/services/proofGenerationServices.js [hunk 1]
```diff
@@ -142,46 +135,39 @@ export async function generateExitPayload(burnTxHash, eventSignature, isMainnet)
...
+      const lastChildBlock = await maticClient.exitUtil.rootChain.getLastChildBlock()
 
       // fetch transaction receipt
-      const receipt = await matic.web3Client.getMaticWeb3().eth.getTransactionReceipt(burnTxHash)
-
-      if (!receipt || !receipt.blockNumber) {
-      // temporary fix for when we receive null receipts
-      // should be reverted back when issue is fixed
+      try {
+        const receipt = await maticClient.client.child.getTransactionReceipt(burnTxHash)
```
Comment (ujjwalgupta94) [inline src/services/proofGenerationServices.js:152]: here you can use `isCheckPointed` API.https://maticnetwork.github.io/matic.js/docs/pos/is-check-pointed/

File: src/services/proofGenerationServices.js [hunk 2]
```diff
@@ -23,29 +22,29 @@ export async function isBlockIncluded(blockNumber, isMainnet) {
...
     try {
       // initialize matic client
-      await initMatic(isMainnet, maticRPC[rpcIndex], ethereumRPC[rpcIndex]).then((obj) => {
-        matic = obj.matic
-      })
+      const maticClient = await initMatic(isMainnet, maticRPC[rpcIndex], ethereumRPC[rpcIndex])
 
       // check last child block included
-      const lastChildBlock = await matic.rootChain.getLastChildBlock()
+      const lastChildBlock = await maticClient.exitUtil.rootChain.getLastChildBlock()
```
Comment (ujjwalgupta94) [inline src/services/proofGenerationServices.js:35]: rootChain is being used multiple times, so can be stored in a variable.

File: src/services/proofGenerationServices.js (hunk 1 above)
Comment (py-zoid) [reply src/services/proofGenerationServices.js:152]: Needed to differentiate between the receiving null receipt for a correct txHash and trying to fetch a wrong txHash to implement the retry logic.

File: src/services/proofGenerationServices.js (hunk 2 above)
Comment (py-zoid) [reply src/services/proofGenerationServices.js]: Changes done

**Review by ujjwalgupta94 — APPROVED** [review]

**Pull request merged**